java/ql/lib/semmle/code/java/PrettyPrintAst.qll

```diff
@@ -742,11 +745,19 @@ private class PpSwitchStmt extends PpAst, SwitchStmt {
   }
 }
 
+private predicate isNonNullDefaultCase(SwitchCase sc) {
+  sc instanceof DefaultCase and not sc instanceof NullDefaultCase
+}
+
 private class PpSwitchCase extends PpAst, SwitchCase {
+  PpSwitchCase() { not this instanceof PatternCase }
+
   override string getPart(int i) {
-    i = 0 and result = "default" and this instanceof DefaultCase
+    i = 0 and result = "default" and isNonNullDefaultCase(this)
+    or
+    i = 0 and result = "case " and not isNonNullDefaultCase(this)
```

Comment:
Right, the problem here is that `NullDefaultCase` isn't a `ConstCase`, so we don't get the `null` or the `", "` printed. Let's just handle the `NullDefaultCase` with a `i = 0 and result = "case null, default"`. Then we can also delete the `getCaseDefaultOffset()`.